source/blog/2017-03-14-a-tale-of-tempest-rpm-with-installers.html.md

```diff
@@ -0,0 +1,70 @@
+---
+title: A tale of Tempest RPM with Installers
+author: chandankumar
+tags: openstack,rdo,tempest,python-tempestconf
+date: 2017-03-14 18:27:02 IST
+---
+
+[Tempest](https://docs.openstack.org/developer/tempest/overview.html) is a set of integration tests to run against OpenStack Cloud.
+Delivering robust and working OpenStack cloud is always challenging. To make sure what we deliver in RDO is robust, we use Tempest
+to perform a set of API and scenario tests against a working cloud using different installers like puppet-openstack-integration, packstack, and tripleo-quickstart. And, It is the story of RDO tempest rpm and how we have done the integration of Tempest rpm with installers so
+that they can be consumed in different CI to deliver a robust cloud.
+
+And the story begins from here:
+
+In RDO, we deliver Tempest as an RPM to be consumed by anyone to test their cloud. Till newton release,
+we maintained a fork of [Tempest](https://github.com/redhat-openstack/tempest) which contains config_tempest.py script to auto generate tempest.conf
+for your cloud and a set of helper scripts to run tempest tests as well as with some backports for each release. We are using tempest rpm which having
+forked tempest as a source till Newton release. From ocata, we have changed the source of Tempest rpm from forked tempest to upstream tempest by keeping
+the old source till newton in RDO through [rdoinfo](https://review.rdoproject.org/r/#/q/topic:switch-upstream-tempest). We are using rdo-patches branch
+to maintain patches backports starting from ocata release.
+
+With this change, we have moved config_tempest.py script from forked tempest to separate project [python-tempestconf](https://github.com/redhat-openstack/python-tempestconf)
```

Comment:
"With this change, we have moved config_tempest.py script from forked tempest to separate project" ->
"With this change, we have moved the config_tempest.py script from the forked tempest (repository) to a separate project"